src/LiveComponent/src/Twig/LiveComponentRuntime.php

```diff
@@ -25,19 +25,21 @@
 final class LiveComponentRuntime
 {
     private LiveComponentHydrator $hydrator;
+    private ComponentFactory $factory;
     private UrlGeneratorInterface $urlGenerator;
     private ?CsrfTokenManagerInterface $csrfTokenManager;
 
-    public function __construct(LiveComponentHydrator $hydrator, UrlGeneratorInterface $urlGenerator, CsrfTokenManagerInterface $csrfTokenManager = null)
+    public function __construct(LiveComponentHydrator $hydrator, ComponentFactory $factory, UrlGeneratorInterface $urlGenerator, CsrfTokenManagerInterface $csrfTokenManager = null)
     {
         $this->hydrator = $hydrator;
+        $this->factory = $factory;
         $this->urlGenerator = $urlGenerator;
         $this->csrfTokenManager = $csrfTokenManager;
     }
 
     public function renderLiveAttributes(Environment $env, object $component): string
     {
-        $url = $this->urlGenerator->generate('live_component', ['component' => AsLiveComponent::forClass($component::class)->getName()]);
+        $url = $this->urlGenerator->generate('live_component', ['component' => $this->nameFor($component)]);
```

Comment:
The one thing that is unfortunate is, right now, _live components_ can only be registered _once per class_. If you have `MarkdownComponent` registered twice as `markdown1` and `markdown2`. When using these twig functions, only the _first one found_'s config will be used. This is because these functions need to look up the _component name_ from the class (as only the component's object is passed).

One solution could be to allow these twig functions to have a second, optional `$name` argument. If passed, it is used. I'm open to other ideas!

Reply:
As an example of where this could be a problem: say you used a 3rd party bundle that registered a `MarkdownComponent` with name/key `markdown` and template `path/to/some/template.html.twig`. You need an _additional implementation_ where you customize the template so you register it manually (same class but with different service name) with name/key `app_markdown` and template `components/app_markdown.html.twig`. When using these twig functions, the bundle's config will always be loaded for this component.

If you completely override the service, _I believe_ you'd be ok - this would only be a problem if you need both implementations...

So maybe an obscure issue that no one will ever come across?

Reply:
I understand the issue. I can't think of any other workaround than allowing a `name` option to be specified. It's pretty simple: if we pass a component object to the Twig function... and that object could potentially have 2 different configs, there's just no way for us to choose the correct one without additional info.

If adding a `$name` argument isn't too much trouble, it would be nice to be pre-planning for this edge-case.

Reply:
I added this. I opted to not document as I believe this is a very obscure edge case but let me know if you'd like documentation now.

My solution is a bit ugly - a lot of logic added to `ComponentFactory::configFor()` that isn't really necessary for the twig component itself. A possible improvement could be to just have `ComponentFactory::getConfig()` that returns the entire config array and let places that require it add their own error checking. Not sure if it's worth it though.